src/tests/ftest/soak/soak.py

```diff
@@ -0,0 +1,338 @@
+
+        # workdir was not successful, not sure why right now
+        self.tmpdir = self.basepath + "/install/tmp"
+        try:
+            os.makedirs(self.tmpdir)
+        except:
+            pass
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        # start the servers
+        self.hostlist_servers = self.params.get("daos_servers",
+                                                '/run/hosts/test_machines/*')
+        filename = WriteHostFile.WriteHostFile(self.hostlist_servers,
+                                               self.workdir)
+        self.server_group = self.params.get("server_group", '/server/',
+                                            'daos_server')
+        print("Servers {} group {} basepath {}".format(self.hostlist_servers,
+                                                       self.server_group,
+                                                       self.basepath))
+        ServerUtils.runServer(filename, self.server_group, self.basepath)
+
+        # setup the storage
+        self.create_pool()
+
+    def tearDown(self):
+        ServerUtils.stopServer(hosts=self.hostlist_servers)
+
+    def test_soak_1(self):
+        """
+        Test ID: DAOS-2192
+        Test Description: This test runs 2 DAOS API IOR jobs.
+        :avocado: tags=soak1
+        """
+
+        try:
+            # retrieve job parameters
+            s1_job1_name = self.params.get("name", '/run/job1/')
+            s1_job1_nodes = self.params.get("nodes", '/run/job1/')
+            s1_job1_processes = self.params.get("process_per_node",
+                                                '/run/job1/')
+            s1_job1_spec = self.params.get("jobspec", '/run/job1/')
+            cmd = [self.get_ior_job(s1_job1_spec, s1_job1_processes,
+                                    s1_job1_name)]
+
+            # write out a slurm job script
+            output = os.path.join(self.tmpdir, s1_job1_name + "_results.out")
+            script1 = slurm_utils.write_slurm_script(self.tmpdir, s1_job1_name,
+                                                    output,
+                                                    int(s1_job1_nodes), cmd)
+            # queue it up to run and register a callback to retrieve results
+            job_id = slurm_utils.run_slurm_script(script1)
+            slurm_utils.register_for_job_results(job_id, job_done, maxwait=3600)
+
+            # queue up a second job
+            s1_job2_name = self.params.get("name", '/run/job2/')
+            s1_job2_nodes = self.params.get("nodes", '/run/job2/')
+            s1_job2_processes = self.params.get("process_per_node",
+                                                '/run/job2/')
+            s1_job2_spec = self.params.get("jobspec", '/run/job2/')
+            cmd = [self.get_ior_job(s1_job2_spec, s1_job2_processes,
+                                    s1_job2_name)]
```

Comment:
Looks like a duplication of lines 178-184.  Can this be encapsulated into a nested function?

Reply:
okay, done